components/builder-jobsrv/src/main.rs

```diff
@@ -36,12 +36,13 @@ fn main() {
     env_logger::init().unwrap();
     let matches = app().get_matches();
     debug!("CLI matches: {:?}", matches);
-    let config = match config_from_args(&matches) {
-        Ok(result) => result,
+    let (subcmd, mut config) = match subcmd_and_config_from_args(&matches) {
+        Ok((s, c)) => (s, c),
         Err(e) => return exit_with(e, 1),
     };
+    config.action = subcmd;
```

Comment:
If subcmd == "migrate" we can just do a Datastore::new(..), datastore.setup()? and then immediately exit - that saves having to do the extra plumbing to execute the setup much deeper in the code.

Reply:
I know it looks like that's the case, and that's the route that I originally went down, but it turns out that's not really true, because of this:

https://github.com/habitat-sh/habitat/blob/master/components/builder-originsrv/src/server/mod.rs#L127

The `DataStore` requires that a `router_pipe` be passed into it, and if you trace through the code to find out what exactly that is, you end up here:

https://github.com/habitat-sh/habitat/blob/master/components/net/src/app/mod.rs#L370

`app_init` is part of an abstraction in the `net` crate and it felt like a Bad Thing™ to just copy/paste the pipe string into the `DataStore::new()` function.  I ended up at the solution that I have here (putting the action as part of the `Config`) because all of the start-up mechanics of our servers are abstracted away, which doesn't lend itself to things like adding parameters to functions.

Reply:
Thinking about it more now, I guess if we're just going to run migrations and exit, I could probably just pass any arbitrary string as the `router_pipe` parameter, which would avoid the need to plumb everything through the way I did.  Mostly I just wanted to avoid breaking things.

Reply:
Ah, I was looking at the jobsrv Datastore which does not need a router_pipe passed in.  It looks like its only needed for sending DB async events, which shouldn't be required for migration. So a bit more refactoring might be required to decouple the Datastore from router_pipe, but that might be a good thing to do anyways.